src/tcptransport.ts

```diff
@@ -43,8 +45,27 @@ export class TCPTransport implements Transport {
                 this.destroy();
             }
             let connected = false;
+            let to: NodeJS.Timeout | undefined;
+            if(timeout) {
+                this.dialTime = Date.now();
```

Comment:
Have this use of dialTime be local, such that dialTime only stores a duration?

Reply:
it is the initial value, that then gets used for the offset. At this point, there's no duration. When it completes we have a duration that we can work from.

Reply:
done.